sdk/dotnet/sdk/Dagger.SDK/GraphQL/Types.cs

```diff
@@ -5,86 +5,114 @@ namespace Dagger.SDK.GraphQL;
 
 public abstract class Value
 {
-    public abstract Task<string> Format();
+    public abstract Task<string> FormatAsync(CancellationToken cancellationToken = default);
 }
 
-public class IdValue<TId>(IId<TId> value) : Value
-    where TId : Scalar
+public class IdValue<TId>(IId<TId> value) : Value where TId : Scalar
 {
-    public override async Task<string> Format()
+    public override async Task<string> FormatAsync(CancellationToken cancellationToken = default)
     {
-        return await new StringValue((await value.Id()).Value).Format();
+        var id = await value.IdAsync(cancellationToken);
+        return await new StringValue(id.Value).FormatAsync(cancellationToken);
     }
 }
 
 public class StringValue(string value) : Value
 {
-    public override Task<string> Format()
+    public override Task<string> FormatAsync(CancellationToken cancellationToken = default)
     {
-        var s = value
```

Comment:
Try to only have one iteration, because Replace iterate over all many time with immutable strings